src/components/ReadingGoal/DeleteReadingGoalModal/index.tsx

```diff
@@ -0,0 +1,92 @@
+import { useState } from 'react';
+
+import Trans from 'next-translate/Trans';
+import useTranslation from 'next-translate/useTranslation';
+
+import styles from './DeleteReadingGoalModal.module.scss';
+
+import Button, { ButtonType, ButtonVariant } from '@/dls/Button/Button';
+import Input from '@/dls/Forms/Input';
+import Modal from '@/dls/Modal/Modal';
+import { deleteReadingGoal } from '@/utils/auth/api';
+import { logButtonClick } from '@/utils/eventLogger';
+
+type DeleteReadingGoalButtonProps = {
+  isDisabled?: boolean;
+};
+
+const DeleteReadingGoalModal = ({ isDisabled }: DeleteReadingGoalButtonProps) => {
```

Comment:
This component is not being used, we should add it.

Reply:
Yes, let's find a good place for it in the UI and add it insha'Allah

Reply:
Okay sounds good.